library/library.mli

```diff
@@ -70,8 +65,17 @@ exception LibUnmappedDir
 exception LibNotFound
 type library_location = LibLoaded | LibInPath
 
+type require_mode = RequireOnlyVo | RequireMostRecent
+(** Describes what files should be looked for at require time.
+
+
+    - [RequireOnlyVo] means only look for `.vo` files
+    - [RequireMostRecent] means look for `.vio` and `.vo` files, and if both
+      exist, emit a warning and load the most recent of the two
```

Comment:
The warning is emitted only if the .vio is more recent than the vo.
Otherwise the .vo is loaded, and no warning is emitted.